client/src/tests/query.tsx

```diff
@@ -0,0 +1,42 @@
+import React from "react";
+import { Query, QuerySpec } from "../query";
+
+interface Result {
+  output: string;
+}
+
+interface Variables {
+  y: string;
+}
+
+const exampleQuery: QuerySpec<Result, Variables> = {
+  query: "query test {}",
+};
+
+const ComponentWithQueryAndVariables: React.SFC<{}> = () => {
+  return (
+    <Query<Result, Variables> query="test" variables={{ y: "yes" }}>
+      {data => (
+        <div>
+          {data.state === "subscribed" ? data.value.output : "loading..."}
+        </div>
+      )}
+    </Query>
+  );
+};
+
+const instanceWithQueryAndVariables = <ComponentWithQueryAndVariables />;
+
+const ComponentWithSpec: React.SFC<{}> = () => {
+  return (
+    <Query query={exampleQuery} variables={{ y: "yes" }}>
+      {data => (
+        <div>
+          {data.state === "subscribed" ? data.value.output : "loading..."}
```

Comment:
i'm not actually sure how it's able to pull the type here... the query does not reference the output or input types at all, except implicitly via `exampleQuery`.

Reply:
ah it does do the inline! that's great

Reply:
this also sure is a lot of indentation